### ISSUE TITLE
Graph Merge Example in ReadMe is not same in Ruby v3.2

### ISSUE THREAD

**Issue by MadBomber** [rewindio/dagwood#7]
The results that I'm getting with Ruby v3.2 for the graph merge example is not the same as is shown in the README document.

```
require 'amazing_print'
require 'dagwood'

recipe1 = Dagwood::DependencyGraph.new(
  add_mustard:      [:slice_bread], 
  add_smoked_meat:  [:slice_bread], 
  close_sandwich:   [:add_smoked_meat]
)

recipe2 = Dagwood::DependencyGraph.new(
  add_mayo:       [:slice_bread], 
  add_turkey:     [:slice_bread], 
  close_sandwich: [:add_turkey, :add_pickles]
)

ultimate_recipe = recipe1.merge(recipe2)

ap ultimate_recipe.order

=begin
[
    [0] :slice_bread,
    [1] :add_mustard,
    [2] :add_smoked_meat,
    [3] :add_pickles,
    [4] :add_turkey,
    [5] :close_sandwich,
    [6] :add_mayo
]
=end
```

Looks messy to me .... and I do not like mayo!

**Comment by smandegar** [comment]
Hey @MadBomber, good catch 🤦 

It seems we need to be more explicit about our `close_sandwich` dependencies - closing the sandwich depends on having some mayo (even if you don't like it!). 

The reason why the individual recipes order properly is due to the order that the ingredients were written in. If I were to place the mayo at the bottom, you'd see a terrible sandwich:

```
bad_sandwich = Dagwood::DependencyGraph.new(
  add_turkey:     [:slice_bread], 
  close_sandwich: [:add_turkey, :add_pickles],
  add_mayo:       [:slice_bread]
)

bad_sandwich.order
=> [:slice_bread, :add_turkey, :add_pickles, :close_sandwich, :add_mayo]
```

but if my `close_sandwich` depended on the mayo:
```
decent_sandwich = Dagwood::DependencyGraph.new(
  add_turkey:     [:slice_bread], 
  close_sandwich: [:add_turkey, :add_pickles, :add_mayo],
  add_mayo:       [:slice_bread]
)

decent_sandwich.order
=> [:slice_bread, :add_turkey, :add_mayo, :add_pickles, :close_sandwich]
```
you'd have a pretty good sandwich. 

I'll update the README to make the `close_sandwich` dependencies more explicit which will resolve the issue with the example.

Thanks for flagging this!

**Comment by smandegar** [comment]
Resolved via https://github.com/rewindio/dagwood/pull/9

**Comment by MadBomber** [comment]
you are welcome. I’m looking forward to making use of your library in my new sandwich shop!  i’ve been thinking about adding a graph viz output to your library. Any objections?Dewayne   o-*On Jan 25, 2023, at 2:14 PM, Sean Mandegar ***@***.***> wrote:﻿
Closed #7 as completed.
